### ISSUE TITLE
"flb_input_chunk.c:1875 errno=0] No error" when using input plugin

### ISSUE THREAD

**Issue by pfrcks** [chronosphereio/calyptia-plugin#55]
Using the input plugin results in multiple errors of the form

[2023/10/19 23:15:23] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error

Here's the config:
```
[INPUT]
    Name containerinventory
    Tag  oneagent.containerInsights.CONTAINER_INVENTORY_BLOB
    run_interval 60
 
[OUTPUT]
    Name stdout
    Match *
```

Here are the logs:
```
[2023/10/19 23:15:22] [ info] [fluent bit] version=2.0.14, commit=bf20c16ea4, pid=33200
[2023/10/19 23:15:22] [debug] [engine] coroutine stack size: 98302 bytes (96.0K)
[2023/10/19 23:15:22] [ info] [storage] ver=1.4.0, type=memory, sync=normal, checksum=off, max_chunks_up=128
[2023/10/19 23:15:22] [ info] [cmetrics] version=0.6.1
[2023/10/19 23:15:22] [ info] [ctraces ] version=0.3.1
[2023/10/19 23:15:22] [ info] [input:containerinventory:containerinventory.0] initializing
[2023/10/19 23:15:22] [ info] [input:containerinventory:containerinventory.0] storage_strategy='memory' (memory only)
[2023/10/19 23:15:22] [ info] [input:containerinventory:containerinventory.0] thread instance initialized
[2023/10/19 23:15:22] [debug] [containerinventory:containerinventory.0] created event channels: read=956 write=960
[2023/10/19 23:15:22] [debug] [input:containerinventory:containerinventory.0] [thread init] initialization OK
[2023/10/19 23:15:22] [debug] [stdout:stdout.0] created event channels: read=972 write=976
[2023/10/19 23:15:22] [ info] [sp] stream processor started
[2023/10/19 23:15:22] [ info] [output:stdout:stdout.0] worker #0 started
[2023/10/19 23:15:23] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:24] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:25] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:26] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:27] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:28] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:29] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:30] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:31] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:32] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:33] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:34] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:35] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:36] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:37] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:38] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:39] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:40] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:41] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:42] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:43] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:44] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:45] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:46] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:47] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:48] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:49] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:50] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:51] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:52] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:53] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:54] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:55] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:56] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:57] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:58] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:15:59] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:00] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:01] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:02] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:03] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:04] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:05] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:06] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:07] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:08] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:09] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:10] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:11] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:12] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:13] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:14] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:15] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:16] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:17] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:18] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:19] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:20] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:21] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:22] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:23] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:24] [debug] [input chunk] update output instances with new chunk size diff=6795
[2023/10/19 23:16:25] [debug] [task] created task=00000183A62808A0 id=0 OK
[2023/10/19 23:16:25] [debug] [output:stdout:stdout.0] task_id=0 assigned to thread #0
[2023/10/19 23:16:25] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:25] [debug] [out flush] cb_destroy coro_id=0
[2023/10/19 23:16:25] [debug] [task] destroy task=00000183A62808A0 (task_id=0)
[2023/10/19 23:16:26] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:27] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:28] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:29] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:30] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:31] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:32] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:33] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:34] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:35] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:36] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:37] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:38] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:39] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:40] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:41] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:42] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:43] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:44] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
[2023/10/19 23:16:45] [error] [C:\projects\fluent-bit-2e87g\src\flb_input_chunk.c:1875 errno=0] No error
```

Happens in both Linux and Windows environments. Fluent-bit version 2.0.14.

containerinventory input plugin is essentially

```
func (p containerInventoryPlugin) Collect(ctx context.Context, ch chan<- plugin.Message) error {
	tick := time.NewTicker(time.Duration(p.runInterval) * time.Second)

	for {
		select {
		case <-ctx.Done():
			err := ctx.Err()
			if err != nil && !errors.Is(err, context.Canceled) {
				return err
			}

			return nil
		case <-tick.C:
			messages := p.enumerate()
			ch <- plugin.Message{
				Record: map[string]any {
					"tag":     tag,
					"messages": messages,
				},
				Time: emitTime,
			}
		}
	}
}
```

where enumerate is a helper function to get records.

**Comment by cosmo0920** [comment]
This is fluent-bit issue and fixed in v2.1.10: https://github.com/fluent/fluent-bit/commit/85be089da9296a0ad92045763044eb8bb63a4449

Could you upgrade using fluent-bit to v2.1.10?

**Comment by niedbalski** [comment]
@pfrcks is this solved? seems so with 2.1.10
